src/Tasks/CultureInfoCache.cs

```diff
@@ -55,7 +57,35 @@ static CultureInfoCache()
         /// <returns>True if the culture is determined to be valid.</returns>
         internal static bool IsValidCultureString(string name)
         {
-            return ValidCultureNames.Contains(name);
+            var isValid = ValidCultureNames.Contains(name);
+            if (isValid)
+                return true;
+
+            var isInvalid = InvalidCultureNames.Contains(name);
+            if (isInvalid)
+                return false;
+
+            CultureInfo culture;
+            try
+            {
+                culture = new CultureInfo(name);
+            }
+            catch (Exception)
+            {
+                InvalidCultureNames.Add(name);
+                return false;
+            }
+
+            // See https://docs.microsoft.com/en-us/dotnet/api/System.Globalization.CultureInfo.LCID#remarks
+            const int LOCALE_CUSTOM_UNSPECIFIED = 0x1000;
+            if (culture.LCID == LOCALE_CUSTOM_UNSPECIFIED)
```

Comment:
Actually, this could be further discussed. What do we want? Accept *any* valid culture or accept *known* (by the operating system) cultures? If we accept *any* valid culture then we can get rid of this check altogether.